### ISSUE TITLE
30tn8336.lmp desyncs (on Windows, but not on Linux?)

### ISSUE THREAD

**Issue by fabiangreffrath** [chocolate-doom/chocolate-doom#915]
The "30tn8336.lmp" TNT NM Max demo desyncs on MAP03 when run with a Chocolate Doom binary that was built on Windows (with MinGW, that is). Strange, though, is the fact that when I play back the demo with a binary built on Linux (also with GCC) it plays through. NB: The demo is supposed to be run with the TNT.WAD file that Steam provides, it desyncs with the "Vanilla" IWAD file in MAP10 (but is not supposed to desync in MAP03 as seen here).

**Comment by fabiangreffrath** [comment]
PS: the demo plays through with PrBoom+ built in the same environment.

**Comment by fabiangreffrath** [comment]
PPS: It happens with both compilers that I tested
```
$ gcc --version
gcc.exe (Rev3, Built by MSYS2 project) 6.3.0
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ clang --version
clang version 4.0.0 (tags/RELEASE_400/final)
Target: i686-w64-windows-gnu
Thread model: posix
InstalledDir: D:\msys32\mingw32\bin
```

**Comment by fabiangreffrath** [comment]
Just confirmed that it plays through on Linux with this compiler
```
$ gcc --version
gcc (Debian 6.3.0-18) 6.3.0 20170516
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by fabiangreffrath** [comment]
Sorry for the spam, but I have more fun facts to add. This is most likely not a bit-ness problem. The i386 Debian package keeps in sync, whereas today's daily build desyncs at the exact same position in the demo if executed with wine. This position changes, though, if `-zonezero -zonescan` are applied.

**Comment by AXDOOMER** [comment]
Where can I find `30tn8336.lmp`? I haven't been able to find a download link via search engines, Doomworld, on COMPET-N or doomedsda.us.

**Comment by JNechaevsky** [comment]
Here it is: http://doomedsda.us/lmps/944/3/evalln8336.zip

**Comment by AXDOOMER** [comment]
>Here it is: http://doomedsda.us/lmps/944/3/evalln8336.zip

Thanks!

> The demo is supposed to be run with the TNT.WAD file that Steam provides, it desyncs with the "Vanilla" IWAD file in MAP10 (but is not supposed to desync in MAP03 as seen here).

I don't think it's supposed to desync at all. I'm playing the demo with [Doom Vanille](https://github.com/AXDOOMER/doom-vanille) using the TNT IWAD from Stream and it went through MAP10 successfully. By "Vanilla" IWAD, you mean the original IWAD that came with the first version of Final Doom ever distributed? The checksum of my Steam IWAD is the same as the original one [in the DoomWiki](https://doomwiki.org/wiki/TNT.WAD) (`4e158d9953c79ccf97bd0663244cc6b6`). Only the *id Anthology* version made changes to MAP10.

**Comment by njankowski** [comment]
If the desync manifests itself as the player being killed by a chaingunner on MAP03, `-gameversion final` fixes this. The Final Doom teleportation bug is almost unnoticeably important in keeping this demo synchronized, but you can see it with the naked eye.

The runner has posted the complete demo on YouTube, and the link will jump right to the teleport in question: https://youtu.be/e0vvYN_UCe8?t=4m28s

When the runner takes this teleport, they are teleported slightly into the air and drop to the ground, something that does not occur if Final Doom compatibility is not enabled.

Enabling Final Doom compatibility allowed the demo to remain synchronized and run to completion in my brief tests.

**Comment by njankowski** [comment]
To be sure that no coincidental desynchronization is taking place, I did get an MSYS2 installation with one of the mentioned compilers:
> gcc.exe (Rev3, Built by MSYS2 project) 6.3.0

Still, the issue only occurs when `-gameversion final` is not used.

Previous post tested:
`2.3.0` pre-built Windows binary
`3.0.0-beta1` (e334de61580e02c7fce892dd185b5c3fd7ff6e5d) built with MSVC

This post tested:
`3.0.0-beta1` (e334de61580e02c7fce892dd185b5c3fd7ff6e5d) built with `gcc.exe (Rev3, Built by MSYS2 project) 6.3.0`

**Comment by fabiangreffrath** [comment]
Thank you very much for this information. However, I fail to see why it should be necessary to pass `-gameversion final` over to Choco when `tnt.wad` is used as the IWAD?

**Comment by AXDOOMER** [comment]
> Unrelated Edit (@AXDOOMER):

Thanks Fabian! I didn't notice he picked up a key.

**Comment by fabiangreffrath** [comment]
@njankowski Could you please post the MD5Sum of your TNT.WAD file?

**Comment by njankowski** [comment]
It is the non-anthology edition of TNT.WAD with an MD5 of `4e158d9953c79ccf97bd0663244cc6b6`.

To be clear, my command line options were `-iwad tnt_1.9.wad -timedemo 30tn8336.lmp`.
It looks like my non-standard WAD name was causing Chocolate Doom to select DOOM2 for the engine type if I did not provide `-gameversion final`. When I renamed the WAD to `tnt.wad`, the demo ran successfully without providing `-gameversion final`.
